test/NuGet.Core.Tests/NuGet.Versioning.Test/NuGetVersionTest.cs

```diff
@@ -326,7 +326,7 @@ public void ToStringFromStringFormat(string versionString, string specialVersion
             var semVer = new NuGetVersion(version, specialVersion);
 
             // Assert
-            Assert.Equal(expected, String.Format("{0}", semVer));
+            Assert.Equal(expected, string.Format("{0}", semVer));
```

Comment:
```suggestion
            Assert.Equal(expected, semVer.ToString());
```

Test code isn't perf critical, but let's not waste CPU cycles on having to parser `"{0}"`.

Reply:
Since I had other feedback to address relevant to this PR, I went ahead and made this change also. However, just for the record, this has nothing to do with changes I made to `SemanticVersion`, and this test was not affected in any way to my knowledge. There are 15 occurrences of `string.Format` in this Test project, so if there's a perf concern, we should evaluate that wholistically.

Reply:
Changing `String` to `string` was also outside the scope of this PR. If you're going to make "boyscout" improvements, it's not much more effort to go that extra step.

And I disagree with waiting to make small improvements unless we address the issue holistically. If we created an issue to change all instances of `string.Format("{0}", x)` to `x.ToString()`, it would be such a low priority that it will never be scheduled. We don't have to reduce tech debt in isolated PRs, we can improve it little by little.